src/main/java/edu/harvard/iq/dataverse/util/json/JsonPrinter.java

```diff
@@ -467,11 +473,12 @@ public static JsonObjectBuilder json(DatasetVersion dsv, List<String> anonymized
                 .add("sizeOfCollection", dsv.getTermsOfUseAndAccess().getSizeOfCollection())
                 .add("studyCompletion", dsv.getTermsOfUseAndAccess().getStudyCompletion())
                 .add("fileAccessRequest", dsv.getTermsOfUseAndAccess().isFileAccessRequest());
-
-        bld.add("metadataBlocks", (anonymizedFieldTypeNamesList != null) ?
-                jsonByBlocks(dsv.getDatasetFields(), anonymizedFieldTypeNamesList)
-                : jsonByBlocks(dsv.getDatasetFields())
-        );
+        if(includeMetadataBlocks) {
+            bld.add("metadataBlocks", (anonymizedFieldTypeNamesList != null) ?
+                    jsonByBlocks(dsv.getDatasetFields(), anonymizedFieldTypeNamesList)
+                    : jsonByBlocks(dsv.getDatasetFields())
+            );
+        }
         if(returnOwners){
             bld.add("isPartOf", getOwnersFromDvObject(dataset));
         }
```

Comment:
I'm just going to add a comment here on the last line of code.

@johannes-darms how do you feel about adding an API test? It might be nice to see this new boolean as an option in UtilIT methods. Plus, it will give confidence that the code is working properly. 😄 I can help with REST Assured assertions, if need be!

Reply:
Tests have been added! Thanks!